frameworks/CSharp/aspnetcore-corert/PlatformBenchmarks/npgsql.rd.xml

```diff
@@ -0,0 +1,51 @@
+<Directives>
+  <Application>
+    <Assembly Name="Npgsql">
```

Comment:
@MichalStrehovsky this is fixed for 6.0 (see https://github.com/npgsql/npgsql/pull/4095).

BTW when testing just now, I did see warnings around generic infinite recursion (see below), rather than OOM, maybe you implemented the check since we talked?

```
ILC : AOT analysis warning IL1005: Npgsql.Internal.TypeHandlers.MultirangeHandler`1<NpgsqlRange`1<NpgsqlRange`1<NpgsqlRange`1<NpgsqlRange`1<__Canon>>>>>.ValidateAndGetLengthMultirange<NpgsqlRange`1<NpgsqlRange`1<NpgsqlRange`1<NpgsqlRange`1<__Canon>>>>>(IList`1<NpgsqlRange`1<NpgsqlR
ange`1<NpgsqlRange`1<NpgsqlRange`1<NpgsqlRange`1<__Canon>>>>>>,NpgsqlLengthCache&,NpgsqlParameter): Method will always throw because: Failed to load type 'Npgsql.Internal.TypeHandling.NpgsqlTypeHandler' from assembly 'Npgsql, Version=6.0.0.0, Culture=neutral, PublicKeyToken=5d8b90d
52f46fda7' [/home/roji/projects/test/Test.csproj]                                                                                                                                                                                                                                         
```

Reply:
That's great to hear, thank you!

Yup, I did: https://github.com/dotnet/runtimelab/pull/1681. Having the compiler just run out of memory was annoying.

Getting rid of the recursions should hopefully speed up the compilation of anything using Npgsql and make the AOT compiled output a lot smaller. (The recursions detection makes the compiler not crash, but the result is still not great because there's not much that can be done with it.)

Reply:
I did notice the compilation speed difference - but it's even better to be able to get an explicit warning! Thanks!